test/e2e/scale.go

```diff
@@ -396,7 +454,39 @@ outer:
 	return results, kerrors.NewAggregate(errs)
 }
 
-func createClusterAndWaitWorker(ctx context.Context, inputChan <-chan string, resultChan chan<- workResult, wg *sync.WaitGroup, baseTemplate []byte, baseClusterName string, input clusterctl.ApplyCustomClusterTemplateAndWaitInput) {
+type clusterCreator func(ctx context.Context, clusterName string, clusterTemplateYAML []byte)
+
+func clusterCreateAndWaitFactory(input clusterctl.ApplyCustomClusterTemplateAndWaitInput) clusterCreator {
```

Comment:
I would suggest to create an input struct for this func which only contains the fields we actually set. Looks like those are only a very small subset

Reply:
As discussed let's add a godoc comment instead why we can't just pass input through